src/main/java/Duke.java

```diff
@@ -1,10 +1,378 @@
+import java.io.FileWriter;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
```

Comment:
Consider arranging the imports so that the libraries from the same file are grouped together.